### PR TITLE
Use hash/hmac one shots in NTLM

### PR DESCRIPTION
**Pull request by vcsjones**
We can use the one-shot HMAC / Hash implementations in a few places.

### CODE REVIEW

**Review by filipnavara — APPROVED** [review]

File: src/libraries/Common/src/System/Net/NTAuthentication.Managed.cs [hunk 1]
```diff
@@ -444,14 +444,10 @@ private static void makeNtlm2Hash(string domain, string userName, ReadOnlySpan<c
...
-
-                using (var hmac = IncrementalHash.CreateHMAC(HashAlgorithmName.MD5, pwHash))
-                {
-                    // strangely, user is upper case, domain is not.
-                    byte[] blob = Encoding.Unicode.GetBytes(string.Concat(userName.ToUpperInvariant(), domain));
-                    hmac.AppendData(blob);
-                    hmac.GetHashAndReset(hash);
-                }
+                // strangely, user is upper case, domain is not.
+                byte[] blob = Encoding.Unicode.GetBytes(string.Concat(userName.ToUpperInvariant(), domain));
```
Comment (stephentoub) [inline src/libraries/Common/src/System/Net/NTAuthentication.Managed.cs:448]: As a separate change, is perf here critical enough that we should look into avoiding all these allocations? There's a string for the ToUpperInvariant, a string for the concatenation, and a byte[] for the Unicode bytes.

File: src/libraries/Common/src/System/Net/NTAuthentication.Managed.cs [hunk 2]
```diff
@@ -502,16 +498,12 @@ private unsafe void WriteChannelBindingHash(Span<byte> hashBuffer)
             {
                 IntPtr cbtData = _channelBinding.DangerousGetHandle();
                 int cbtDataSize = _channelBinding.Size;
-
-                using (var md5 = IncrementalHash.CreateHash(HashAlgorithmName.MD5))
-                {
-                    md5.AppendData(new Span<byte>((void *)cbtData, cbtDataSize));
-                    md5.GetHashAndReset(hashBuffer);
-                }
+                int written = MD5.HashData(new Span<byte>((void *)cbtData, cbtDataSize), hashBuffer);
```
Comment (stephentoub) [inline src/libraries/Common/src/System/Net/NTAuthentication.Managed.cs:501]:
```suggestion
                int written = MD5.HashData(new Span<byte>((void*)cbtData, cbtDataSize), hashBuffer);
```

**Review by stephentoub — APPROVED** [review]

File: src/libraries/Common/src/System/Net/NTAuthentication.Managed.cs (hunk 1 above)
Comment (vcsjones) [reply src/libraries/Common/src/System/Net/NTAuthentication.Managed.cs:448]: I'll take a look at the code paths for this and see how hot the path is, and make a follow up PR accordingly.

Comment (filipnavara) [reply src/libraries/Common/src/System/Net/NTAuthentication.Managed.cs:448]: It is not a hot path but it would be nice to optimize it.

**Review by wfurt — APPROVED** [review]
LGTM.

File: src/libraries/Common/src/System/Net/NTAuthentication.Managed.cs (hunk 1 above)
Comment (vcsjones) [reply src/libraries/Common/src/System/Net/NTAuthentication.Managed.cs:448]: @filipnavara I'll tackle that and other optimizations once https://github.com/dotnet/runtime/pull/70720 is in.

**Pull request merged**